### DISCUSSION TITLE
QRcode shows but can't connect: verify.walletconnect postMessage error

### DISCUSSION THREAD

**Discussion by webmass** [WalletConnect/walletconnect-monorepo#2594] [Developer Support]
Hey,
Migrating from v1 to v2
Provider: `web3-react`

Similar error message with https://github.com/orgs/WalletConnect/discussions/2482, though in my case the modal does open, but I can't connect.

I get the following error when the wallet connect modal opens:
`Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://verify.walletconnect.com') does not match the recipient window's origin ('http://localhost:3000').`

Sometimes I get `No matching key. expirer: topic:8a778bf0e83906f90c2759483e1c617c026912a56189f54fba76110400f046e6`

When scanning with the metamask app, the scan feature in metamask closes, but connection does not happen with the dapp, no messages.

Connector initialization:
```
export const [walletConnectV2, walletConnectV2Hooks] = initializeConnector<WalletConnectV2>(
  actions =>
    new WalletConnectV2({
      actions,
      options: {
        projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PID!,
        chains: mainChains,
        optionalChains: secondaryChains,
        showQrModal: true
      }
    })
);
```

Connector activation that triggers the opening of the mocal:
`walletConnectV2.activate()`

Node 16+
Using package "@web3-react/walletconnect-v2": "^8.3.2"

**Comment by glitch-txs** [comment]
hi! could you fix it? could you provide a minimal reproduction example repo?

**Reply by webmass** [reply]
Hi
I still have the issue, thought that maybe it was a known issue.

Just created a repo to reproduce it:

https://github.com/webmass/wallet-connect-v2

Extremely simplified code from my production codebase which has the same issue.

**Reply by glitch-txs** [reply]
Ah you were testing with MM, they don't support v2 yet but they will very soon. The issue shouldn't affect the functionalities and I think it probably just needs an update of deps on the web3-react side but not 100% sure. I'd ask them about this

**Reply by webmass** [reply]
Wallets tried:
Trustwallet: scanning QR code works
Metamask: ko
Coinbase wallet: ko

**Reply by finessevanes** [reply]
@webmass what does `ko` mean?

**Reply by webmass** [reply]
ko = not ok, details:

On metamask mobile the scanner ui just closes after scanning the QR code, no connection is made with the dapp, the QR code modal stays open.

On coinbase wallet the scanner ui stays open after scanning the QR code, a message says to try again, no connection is made with the dapp, the QR code modal stays open.

You should be able to reproduce the issue with the provided repo.

**Reply by finessevanes** [reply]
@webmass neither Coinbase nor MetaMask currently support v2, that's why you're unable to connect.

**Reply by webmass** [reply]
> @webmass neither Coinbase nor MetaMask currently support v2, that's why you're unable to connect.

Hmm I see thanks, a bit surprising from them, hopefully they add support a bit before the v1 is actually sunset.
